README.md

```diff
@@ -1,9 +1,73 @@
 This repository contains my LeetCode, CodeWars (Kata) and HackerRank problem solutions which is mostly written with Python. I'll try to keep solving problems in my free time to see different variant of questions and improve my algorithm solving skills.
 
-<br>
+Note: Files with the '-u' at the end means they are unfinished.
+
+## Table of Contents
+
+- [Features](#features)
+- [Getting Started](#getting-started)
+- [Folder Structure](#folder-structure)
+- [Contribution Guidelines](#contribution-guidelines)
+- [License](#license)
+
+## Features
+
+- **Comprehensive Solutions**: Each algorithmic problem is accompanied by a detailed solution, complete with explanations, edge cases, and runtime complexities.
+  
+- **Optimized Algorithms**: We pride ourselves on providing not just any solution, but the most optimized one. Our algorithms are crafted with performance in mind, ensuring that you get the best out of every line of code.
+  
+- **Variety of Problems**: Our repository covers a broad spectrum of problems, from the basics to the most challenging ones. Whether you're a novice or an expert, there's something here for everyone.
+  
+- **Clean and Commented Code**: We believe that the best code is the one that speaks for itself. Our solutions are written in a clean and readable manner, peppered with comments to guide you through.
+
+## Getting Started
+
+1. **Clone the Repository**: 
+   \```bash
+   git clone https://github.com/finn-str/Algorithm-Solutions.git
+   \```
+   
+2. **Navigate and Explore**: Once cloned, navigate through the repository. Each problem has its dedicated folder with the solution and a `README.md` explaining the approach.
+
+## Folder Structure
+
+Our repository is organized based on difficulty levels, ensuring a smooth progression for those looking to learn:
+
+```
+Algorithm-Solutions/
+│
+├── beginner/
+│   ├── problem1/
+│   │   ├── solution.py
+│   │   └── README.md
+│   └── problem2/
+│       ├── solution.js
+│       └── README.md
+│
+├── intermediate/
+│   └── problem1/
+│       ├── solution.java
+│       └── README.md
+│
+└── advanced/
+    └── problem1/
+        ├── solution.cpp
+        └── README.md
+```
+
+> Note: This is a generic folder structure. Replace with the actual structure of the repository.
+
+## Contribution Guidelines
+
+We're always on the lookout for better solutions and optimizations. If you believe you have a more efficient solution or a new algorithmic problem that could benefit the community:
+
+1. Fork the repository.
+2. Make your changes.
+3. Submit a [pull request](https://github.com/finn-str/Algorithm-Solutions/pulls) with a detailed explanation of your solution or problem.
```

Comment:
This link needs to be directed to my repository, like this:
`3. Submit a [pull request](https://github.com/Alperencode/Algorithm-Solutions/pulls) with a detailed explanation of your solution or problem.`